File: prj189_EXCELsort.py
#!/usr/bin/python
#-*- coding:utf-8 -*-

#Author: Ben

class Student:
    def __init__(self, ID, name, grade):
        self.ID = ID
        self.name = name
        self.grade = grade

def myCmp1(a, b):
    return cmp(a.ID, b.ID)

def myCmp2(a, b):
    if a.name != b.name:
        return cmp(a.name, b.name)
    else:
        return cmp(a.ID, b.ID)

def myCmp3(a, b):
    if a.grade != b.grade:
        return cmp(a.grade, b.grade)
    else:
        return cmp(a.ID, b.ID)

idx = 0
while True:
    try:
        idx += 1
        N, C = map(int, raw_input().split())
        if N==0 and C==0:
            break
        students = [0]*N
        for i in range(N):
            ID, name, grade = raw_input().split()
            students[i] = Student(ID, name, int(grade))
        exec('students.sort(myCmp'+str(C)+')')
        print 'Case: %d' % idx
        for stu in students:
            print stu.ID, stu.name, stu.grade
    except:
        break

File: prj189_EXCELsort.cpp
#include <iostream>
#include <algorithm>
#include <string.h>
#include <stdio.h>

using namespace std;

class Student
{
public:
    int ID, grade; char name[9];
    Student() {}
}students[100001];

bool myCmp1(const Student &a, const Student &b)
{
    return a.ID<b.ID;
}

bool myCmp2(const Student &a, const Student &b) {
    int tmp = strcmp(a.name, b.name);
    if(tmp) {
        return tmp<0;
    } else {
        return a.ID<b.ID;
    }
}

bool myCmp3(const Student &a, const Student &b) {
    if(a.grade != b.grade) {
        return a.grade<b.grade;
    } else {
        return a.ID<b.ID;
    }
}

int main()
{
    int N, C, idx=0;
    while(scanf("%d %d", &N, &C) && N!=0) {
        idx++;
        for(int i=0; i<N; i++) {
            scanf("%d %s %d", &students[i].ID, students[i].name, &students[i].grade);
        }
        if(C==1) {
            sort(students, students+N, myCmp1);
        } else if(C==2) {
            sort(students, students+N, myCmp2);
        } else {
            sort(students, students+N, myCmp3);
        }
        printf("Case %d:\n", idx);
        for(int i=0; i<N; i++) {
            printf("%06d %s %d\n", students[i].ID, students[i].name, students[i].grade);
        }
    }
    return 0;
}
